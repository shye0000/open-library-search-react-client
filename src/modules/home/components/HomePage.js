import React from 'react';
import {connect} from 'react-redux';
import './HomePage.scss';

export class HomePage extends React.Component {

	render() {
		return (
			<div className="home">
				home page
			</div>
		);
	}
}

const mapStateToProps = () => {
	return {};
};

const mapDispatchToProps = () => {
	return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
