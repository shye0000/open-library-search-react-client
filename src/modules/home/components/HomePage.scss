@import "../../../styles/vars";
@import "../../../styles/colors";
.home {
  .search-wrapper {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: $sm-padding $sm-padding 0 $sm-padding;
    &.searched {
      position: relative;
      top: 0;
      left: 0;
      width: 100%;
      transform: none;
      border-bottom: 1px solid $light-grey;
      box-shadow: 0 0 11px rgba(0, 0, 0, 0.23);
    }
  }
  .results-wrapper {
    padding: $lg-padding $sm-padding;
  }
}