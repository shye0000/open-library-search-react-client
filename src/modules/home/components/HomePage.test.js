import React from 'react';
import { HomePage } from './HomePage';
import { shallow } from 'enzyme';

describe('HomePage', () => {

	it('render correctly', () => {
		const props = {
			fetching: false,
			fetchSuccess: false,
			root: null
		};
		const wrapper = shallow(<HomePage {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

});