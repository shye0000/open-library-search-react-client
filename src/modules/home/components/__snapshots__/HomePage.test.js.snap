// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    fetchSuccess={false}
    fetching={false}
    root={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-wrapper"
        >
          <Connect(Form(Search)) />
        </div>,
        null,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Form(Search)) />,
          "className": "search-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-wrapper"
          >
            <Connect(Form(Search)) />
          </div>,
          null,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Form(Search)) />,
            "className": "search-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
